src/cargo/core/compiler/custom_build.rs

```diff
@@ -715,6 +722,24 @@ impl BuildOutput {
         const DOCS_LINK_SUGGESTION: &str = "See https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script \
                 for more information about build script outputs.";
 
+        fn check_minimum_supported_rust_version_for_new_syntax(
+            msrv: &Option<RustVersion>,
+        ) -> CargoResult<()> {
+            let new_syntax_added_in = &RustVersion::from_str("1.76.0")?;
+
+            if let Some(msrv) = msrv {
+                if msrv < new_syntax_added_in {
+                    bail!(
+                        "the `cargo::` syntax for build script output instructions was added in \
+                        Rust 1.76.0, but the minimum supported Rust version is {msrv}.\n\
+                        {DOCS_LINK_SUGGESTION}"
```

Comment:
we should name the target to be clear this is referring to that specific package's MSRV

`but {name}'s minimum supported Rust version is {msrv}`

Reply:
Added. Thanks!